### PR TITLE
Throw an error when operationId is missing in client generation

### PR DESCRIPTION
**Pull request by aneeshafedo**
Resolves https://github.com/ballerina-platform/ballerina-openapi/issues/491

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi-cli/src/main/java/io/ballerina/openapi/cmd/CodeGenerator.java [hunk 1]
```diff
@@ -625,6 +633,27 @@ public OpenAPI normalizeOpenAPI(Path openAPIPath) throws IOException, BallerinaO
...
+     */
+    private void validateOperationIds(Set<Map.Entry<String, PathItem>> paths) throws BallerinaOpenApiException {
+        for (Map.Entry<String, PathItem> entry: paths) {
+            for (Operation operation : entry.getValue().readOperations()) {
+                if (operation.getOperationId() != null) {
+                    String operationId = getValidName(operation.getOperationId(), false);
+                    operation.setOperationId(operationId);
+                } else {
+                    throw new BallerinaOpenApiException("Operation Id is missing. " +
+                            "Please add Operation Id for each operation available. ");
```
[the remark this answers is not in the document]
Comment (indikasampath2000) [reply openapi-cli/src/main/java/io/ballerina/openapi/cmd/CodeGenerator.java:651]: Can we be specific about what resource missing the operationId?
`OperationId is missing for the resource path:  + resourcePath`

[the remark this answers is not in the document]
Comment (aneeshafedo) [reply openapi-cli/src/main/java/io/ballerina/openapi/cmd/CodeGenerator.java]: Updated

**Review by indikasampath2000 — APPROVED** [review]

File: openapi-cli/src/test/java/io/ballerina/openapi/generators/schema/PrimitiveDataTypeTests.java [hunk 2]
```diff
@@ -82,7 +82,7 @@ public void generateMissingDatatype() throws IOException, BallerinaOpenApiExcept
     @Test(description = "When the component schema has primitive data type instead of object schema")
     public void generateSchemaForPrimitiveData() throws IOException, BallerinaOpenApiException {
         Path definitionPath = RES_DIR.resolve("swagger/schema_with_primitive.yaml");
-        OpenAPI openAPI = codeGenerator.normalizeOpenAPI(definitionPath);
+        OpenAPI openAPI = codeGenerator.normalizeOpenAPI(definitionPath, true);
```
Comment (abeykoon) [inline openapi-cli/src/test/java/io/ballerina/openapi/generators/schema/PrimitiveDataTypeTests.java:85]: What does this second parameter `true` does?

Comment (aneeshafedo) [reply openapi-cli/src/test/java/io/ballerina/openapi/generators/schema/PrimitiveDataTypeTests.java:85]: if `true` is given, new operationIds will not be generated for missing operationIds.

**Pull request merged**